fractions/jose/src/main/java/org/wildfly/swarm/jose/provider/JoseFactory.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2015-2016 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.swarm.jose.provider;
+
+import java.net.URL;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ServiceLoader;
+
+import org.jboss.logging.Logger;
+import org.wildfly.swarm.jose.Jose;
+import org.wildfly.swarm.jose.JoseConfiguration;
+
+public abstract class JoseFactory {
+
+    private static Logger log = Logger.getLogger(JoseFactory.class);
+    private static JoseFactory instance;
+
+    /**
+     * Obtain the JoseFactory using the ServiceLoader pattern.
+     *
+     * @return the factory instance
+     */
+    public static JoseFactory instance() {
+        if (instance == null) {
+            synchronized (JoseFactory.class) {
+                if (instance != null) {
```

Comment:
Looks like double-checked locking, which only works if `instance` is `volatile`.

Reply:
Copying and pasting does not always work :-), let me fix it. I'll also do a minor MP-JWT PR